xlators/debug/io-stats/src/io-stats.c

```diff
@@ -1080,9 +1077,10 @@ _io_stats_write_latency_sample(xlator_t *this, ios_sample_t *sample,
         if (!port)
             goto err;
         *port_pos = '\0';
-        hostname = gf_rev_dns_lookup_cached(identifier, conf->dnscache);
-        if (!hostname)
-            hostname = "Unknown";
+        addr = gf_dns_lookup_address_cached(identifier, ctx->dnscache);
+        if (addr)
+            getnameinfo(addr->ai_addr, addr->ai_addrlen, hostname,
```

Comment:
Why not use `addr->ai_canonname` instead of another request ? (`AI_CANONNAME` flag will be needed in getaddrinfo).

Reply:
ai_canonname does not work in case of ip_address, it works only while identifier is a hostname.

Reply:
In that case, why we do need to call `gf_dns_lookup_address_cached()` ? if `identifier` is already a hostname, we don't need to do anything. If it's an IP, we can directly call `getnameinfo()`. Probably we should incorporate `getnameinfo()` into the cached information so that we can resolve from host to IP and from IP to host.